src/modules/blogs/blog.routes.ts

```diff
@@ -0,0 +1,20 @@
+import { Router } from "express";
+import { BlogController } from "./blog.controller";
+import { validateDto } from "../../common/validators";
+import { CreateBlogDto } from "./dto/create-blog.dto";
+import { authenticateToken } from "../../common/middlewares/authenticate-token";
+
+export const BlogRouter = Router();
+
+BlogRouter.post(
+  "/",
+  authenticateToken,
+  validateDto(CreateBlogDto),
+  BlogController.createBlogHandler,
+);
+
+BlogRouter.get("/", authenticateToken, BlogController.getMyBlogsHandler);
+BlogRouter.get("/public", BlogController.getPublicBlogsHandler);
+BlogRouter.get("/:id", authenticateToken, BlogController.getBlogByIdHandler);
+BlogRouter.put("/:id", authenticateToken, BlogController.updateBlogHandler);
```

Comment:
and put one middleware to check whether post is being updated by its own owner. if not reject it.

Reply:
I think ```authenticateToken``` middleware does the same...